doc/reference/security.rst

```diff
@@ -93,8 +91,233 @@ access to the field:
 
 .. todo::
 
-    field access groups apply to administrator in fields_get but not in
+    field access groups apply to the Superuser in fields_get but not in
     read/write...
 
-.. _foo: http://google.com
-.. _time module: https://docs.python.org/2/library/time.html
+.. _time module: https://docs.python.org/3/library/time.html
+
+
+.. _reference/security/pitfalls:
+
+Security Pitfalls
+=================
+
+As a developer, it is important to understand the security mechanisms and avoid
+common mistakes leading to unsecure code.
+
+Unsafe Public Methods
+---------------------
+
+Any public method can be execute via a :ref:`RPC call
+<webservices/odoo/calling_methods>` with the chosen parameters. The methods
+starting with a ``_`` are considered as private in Odoo and not callable from
+an action button or external API.
+
+On public methods, the record on which a method is executed and the parameters
+can not be trusted.
+
+.. code-block:: python
+
+    # this method is public and its arguments can not be trusted
+    def action_done(self):
+        if self.state == "draft" and self.user_has_groups('base.manager'):
+            self._set_state("done")
+
+    # this method is private and can only be called from other python method
+    def _set_state(self, new_state):
+        self.sudo().write({"state": new_state})
+
+Making a method private is obviously not enough and care must be taken to use it
+properly.
+
+Bypassing the ORM
+-----------------
+You should never use the database cursor directly when the ORM can do the same
+thing! By doing so you are bypassing all the ORM features, possibly the
+transactions, access rights and so on.
```

Comment:
nit: probably not the transactions as that's an sql-level thing, however it's bypassing a bunch of automated behaviours like translations, automatic fields, `active` field handling (though sometimes these could be considered misfeatures, YMMV)